### PR TITLE
Support all number types when decoding a point

### PR DESCRIPTION
**Pull request by corylanou**
Fixes https://github.com/influxdb/influxdb/issues/3824

Also removes the legacy need to append a `.0` to float types.  This will improve performance and storage.

### CODE REVIEW

**Comment by corylanou** [comment]
cc @jwilder

File: tsdb/points.go [hunk 1]
```diff
@@ -1309,25 +1308,39 @@ func (p Fields) MarshalBinary() []byte {
...
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
+			b = append(b, 'i')
+		case int32:
+			b = append(b, []byte(strconv.FormatInt(int64(t), 10))...)
 			b = append(b, 'i')
 		case int64:
 			b = append(b, []byte(strconv.FormatInt(t, 10))...)
 			b = append(b, 'i')
+		case uint:
+			b = append(b, []byte(strconv.FormatUint(uint64(t), 10))...)
```
Comment (jwilder) [inline tsdb/points.go]: Not sure if there is a good way to handle this but when we parse things back, we use `strconv.ParseInt`.  I believe we could parse back the wrong value if the `uint` is beyond the max int range.  A good test to try would be to have a field type as a `uint64` with a value of `math.MaxUint64` and format as string and parse back.

**Comment by jwilder** [comment]
Since we don't store `uints`, should the signature for `MarshalBinary` be updated to return an error instead so we prevent some silent errors from happening?  May have some rippling effects in the code though.

File: tsdb/points.go (hunk 1 above)
Comment (corylanou) [inline tsdb/points.go]: Could we check when we parse back if it starts with a `-`?  If it does, it's ParseInt, if not, it's ParseUint.  Would that work?

Comment (jwilder) [inline tsdb/points.go]: I think that would just cause a panic when the shard tries to write: https://github.com/influxdb/influxdb/blob/master/tsdb/shard.go#L509

Comment (corylanou) [inline tsdb/points.go]: Ok, so I think the right fix is to not support uint64.  At this point that using the Go client library if you send a `uint64` in, that it is going to get converted to a string.  If you are using straight line protocol, this won't be an issue, as anything over `int64` will fail one of our checks and error out.

/cc @jwilder @beckettsean

**Comment by jwilder** [comment]
:+1: On green build.

**Pull request merged**